merger.go

```diff
@@ -35,32 +35,27 @@ func (h *MinHeap) Pop() interface{} {
 	return item
 }
 
-func mergeFiles(iterators []*SSTableIterator, file *os.File) error {
-	writer := NewTempWriter(file)
-	defer file.Close()
-
+func (db *DB) mergeIterators(iterators []*SSTableIterator, targetLevel int) ([]*SSTable, error) {
```

Comment:
### Runtime Errors
Handle potential runtime errors when merging iterators. Specifically, ensure that `db.mergeIterators(iterators, targetLevel)` is checked for errors and handle them appropriately. Consider logging the error and returning a more informative error message:
```go
if err != nil {
    log.Printf("Error merging iterators: %v", err)
    return nil, fmt.Errorf("failed to merge iterators: %w", err)
}
```